lib/tesla/mock.ex

```diff
@@ -229,23 +229,12 @@ defmodule Tesla.Mock do
 
   # Gets the mock fun for the current process or its ancestors
   defp pdict_get do
-    pid_holder =
-      Enum.find(Process.get(:"$ancestors", []), self(), fn ancestor ->
-        is_holder? =
-          ancestor
-          |> Process.info()
-          |> Keyword.get(:dictionary)
-          |> Keyword.get(__MODULE__)
-
-        !is_nil(is_holder?)
-      end)
-      |> case do
-        nil -> raise "Unknown pid_holder in mock"
-        pid when is_pid(pid) -> pid
-        name when is_atom(name) -> Process.whereis(name)
-      end
+    potential_mock_holder_pids = [self() | Enum.reverse(Process.get(:"$callers", []))]
```

Comment:
We search for mocks hierarchically. Even though, it is unlikely that we've set mocks both in the caller and in the process, I would be super if I used mock in the current process and it found a mock from somewhere else.

Hence, I start searching from mocks from the current process and then up the callers hierarchy.

In most situations `$callers` and `$ancestors` are the same. E.g. spawning a task, starting a GenServer puts the same pid in both.

The notable difference is starting a supervised task because the ancestors of a supervised task is its supervisor, while the caller is the process that asked to spawn it.

I believe that for mocking `$callers` is the correct one to use.